src/main/java/hudson/plugins/mercurial/HgExe.java

```diff
@@ -176,10 +181,18 @@ public void close() throws IOException, InterruptedException { // TODO implement
     private static ArgumentListBuilder findHgExe(@CheckForNull MercurialInstallation inst, @CheckForNull StandardUsernameCredentials credentials, Node node, TaskListener listener, boolean allowDebug) throws IOException, InterruptedException {
         ArgumentListBuilder b = new ArgumentListBuilder();
         if (inst == null) {
-            b.add(Jenkins.getInstance().getDescriptorByType(MercurialSCM.DescriptorImpl.class).getHgExe());
+            final Jenkins jenkins = Jenkins.getInstance();
+            if (jenkins == null) {
+                throw new IOException("Jenkins instance is not ready");
+            }
+            b.add(jenkins.getDescriptorByType(MercurialSCM.DescriptorImpl.class).getHgExe());
         } else {
             // TODO what about forEnvironment?
-            b.add(inst.executableWithSubstitution(inst.forNode(node, listener).getHome()));
+            final String toolHome = inst.forNode(node, listener).getHome();
+            if (toolHome == null) {
+                throw new IOException("Cannot determine tool home for " + inst);
```

Comment:
I think this is actually not possible for other reasons, but forget details now. Good enough.